web/src/user_group_edit.ts

```diff
@@ -827,29 +897,33 @@ export function change_state(section, left_side_tab, right_side_tab) {
     empty_right_panel();
 }
 
-function compare_by_name(a, b) {
+function compare_by_name(a: UserGroup, b: UserGroup): number {
     return util.strcmp(a.name, b.name);
 }
 
-function redraw_left_panel(tab_name) {
+function redraw_left_panel(tab_name: string): void {
     let groups_list_data;
     if (tab_name === "all-groups") {
         groups_list_data = user_groups.get_realm_user_groups();
     } else if (tab_name === "your-groups") {
         groups_list_data = user_groups.get_user_groups_of_user(people.my_current_user_id());
     }
+    if (groups_list_data === undefined) {
+        return;
+    }
```

Comment:
Under what circumstances do we expect this to be undefined?

Reply:
Oh, I think this isn't meant to be possible, because there are only two tabs? Probably better to have an assert, but best to just specify the type of `tab_name` as a union of those two values.

Reply:
I think that's the right fix but it's likely a bit involved as a refactor to do this across the whole file (and maybe stream_edit.ts too) and thus a good follow-up.